contentctl/actions/inspect.py

```diff
@@ -179,82 +183,173 @@ def inspectAppCLI(self, config:inspect)-> None:
             json.dump(j, logfile, indent=3, )
-            
+
         reports = j.get("reports", [])
         if len(reports) != 1:
             raise Exception("Expected to find one appinspect report but found 0")
         verbose_errors = []
-        
+
         for group in reports[0].get("groups", []):
-            for check in group.get("checks",[]):
-                if check.get("result","") in status_types:                                    
+            for check in group.get("checks", []):
+                if check.get("result", "") in status_types:
                     verbose_errors.append(f" - {check.get('result','')} [{group.get('name','NONAME')}: {check.get('name', 'NONAME')}]")
         verbose_errors.sort()
-        
+
         summary = j.get("summary", None)
         if summary is None:
             raise Exception("Missing summary from appinspect report")
         msgs = []
         generated_exception = False
         for key in status_types:
-            if summary.get(key,0)>0:
+            if summary.get(key, 0) > 0:
                 msgs.append(f" - {summary.get(key,0)} {key}s")
                 if key in exception_types:
                     generated_exception = True
-        if len(msgs)>0 or len(verbose_errors):
+        if len(msgs) > 0 or len(verbose_errors):
             summary = '\n'.join(msgs)
             details = '\n'.join(verbose_errors)
             summary = f"{summary}\nDetails:\n{details}"
             if generated_exception:
-                raise Exception(f"AppInspect found [{','.join(exception_types)}] that MUST be addressed to pass AppInspect API:\n{summary}")        
+                raise Exception(f"AppInspect found [{','.join(exception_types)}] that MUST be addressed to pass AppInspect API:\n{summary}")
             else:
-                print(f"AppInspect found [{','.join(status_types)}] that MAY cause a failure during AppInspect API:\n{summary}")            
+                print(f"AppInspect found [{','.join(status_types)}] that MAY cause a failure during AppInspect API:\n{summary}")
         else:
             print("AppInspect was successful!")
-                
+
         return
+
+    def check_detection_metadata(self, config: inspect) -> None:
+        """
+        Using a previous build, compare the savedsearches.conf files to detect any issues w/
+        detection metadata.
+
+        :param config: an inspect config
+        :type config: :class:`contentctl.objects.config.inspect`
+        """
+        # TODO (#282): We should be inspect the same artifact we're passing around from the
+        #   build stage ideally
+        # Unpack the savedsearch.conf of each ESCU package
+        current_build_conf = SavedsearchesConf.init_from_package(
+            package_path=config.getPackageFilePath(include_version=False),
+            app_name=config.app.label,
+            appid=config.app.appid
+        )
+        previous_build_conf = SavedsearchesConf.init_from_package(
```

Comment:
@pyth0n1c  in our offline discussion we thought we would indicate a need to bump ALL detections in the case that the metadata could not be parsed from the stanza in a prior build due to some intervening change in metadata structure.

This poses issues in that:
* We detect the parsing issue and tell the user they need to update all detection versions (easy!)
* The user re-runs inspect (manually or in the CI) but inspect spits out the same message that they need to bump detection versions. `inspect` can't parse the versions from a prior build for comparison to see they've been bumped, and so it will endlessly report an issue until the next release is up on Splunkbase

So, instead of this, I suggest we just allow extra values to be parsed in the DetectionMetadata. I've added a note in the portion of `DetectionAbstract` which generates the metadata field for the conf, which hightlights that ADDITIVE operations to this metadata dict are fine, but removing any fields or changing types will cause breaking issues in inspect.

If we are expected to start validating against new fields in the metadata in the future, we can address that when the time comes (possibly making these new fields optional for backwards compatibility).

Thoughts?